Fix check for conf file ownership

ArangoGutierrez opened the pull request:
**Description of the Pull Request (PR):**

Add fix on the check for ownership of config file, move check from
config.Parser() to actions.go after setuid check.
As stated by @DrDaveD , PR #2533 breaks unprivileged installations, and root-installed installation with the -u option.

Signed-off-by: Eduardo Arango <eduardo@sylabs.io>

**This fixes or addresses the following GitHub issues:**

- Fixes #2550 


**Before submitting a PR, make sure you have done the following:**

- Read the [Guidelines for Contributing](../CONTRIBUTING.md), and this PR conforms to the stated requirements.
- Added changes to the [CHANGELOG](../CHANGELOG.md) if necessary according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added tests to validate this PR and tested this PR locally with a `make testall`
- Based this PR against the appropriate branch according to the [Contribution Guidelines](../CONTRIBUTING.md)
- Added myself as a contributor to the [Contributors File](../CONTRIBUTORS.md)


Attn: @singularity-maintainers

ArangoGutierrez committed:
Fix check for conf file ownership

Add fix on the check for ownership of config file, move check from
config.Parser() to actions.go after setuid check.
As stated by @DrDaveD , PR #2533 breaks unprivileged installations, and root-installed installation with the -u option.

Signed-off-by: Eduardo Arango <eduardo@sylabs.io>

DrDaveD commented on internal/app/singularity/actions.go:
```diff
@@ -553,6 +554,12 @@ func execStarter(cobraCmd *cobra.Command, image string, args []string, name stri
 		}
+
+		// check for ownership of singularity.conf file when !UserNamespace
+		if !fs.IsOwner(configurationFile, 0) {
```
Make this an 'else if' to work correctly when starter-suid is missing.  That's the case in an unprivileged installation that I maintain.

cclerget replied:
@ArangoGutierrez Those checks should definitively go in `PrepareConfig` function (`internal/pkg/runtime/engines/singularity/prepare.go`) to have something like this : 

```
	if starterConfig.GetIsSUID() {
		// check for ownership of singularity.conf
		if !fs.IsOwner(configurationFile, 0) {
			sylog.Fatalf("%s must be owned by root", configurationFile)
		}
	}
```

Same for `capability.json` but in `prepareUserCaps` function.

CLI won't prevent `starter-suid` from reading a configuration file not owned by root, that's why it must be done in `internal/pkg/runtime/engines/singularity/prepare.go`

DrDaveD reviewed (changes requested):
If I do chown -R on /etc/singularity and run unprivileged, I get the error
```
FATAL:   /etc/singularity/capability.json must be owned by root
```

ArangoGutierrez committed:
fix comment on line 322

Signed-off-by: Eduardo Arango <eduardo@sylabs.io>

ArangoGutierrez committed:
Run ownership checks when SUID is enabled

mv checks to PrepareConfig function in internal/pkg/runtime/engines/singularity/prepare.go

Signed-off-by: Eduardo Arango <eduardo@sylabs.io>

ArangoGutierrez committed:
Merge branch 'master' into check_ownership_config

DrDaveD reviewed (approved):
Looks good to me now.  I tried on el 7.6, with the commits from this PR cherry-picked into a branch of release-3.0.  I tried a chown -R on /etc/singularity with a regular root install and exec -u, and tried an unprivileged install in a scratch directory with libexec/singularity/bin/starter-suid removed.

cclerget reviewed (approved):
LGTM
